### PR TITLE
Azure Disk Template 1.4.0

### PR DESCRIPTION
**Pull request by gauravAggrhari**
## Test Results

**get pods in `kube-system`**
```
% kubectl get pods -n kube-system | grep azuredisk
NAME                                        READY   STATUS    RESTARTS   AGE
csi-azuredisk-controller-849d854b96-6jbjg   5/5     Running   0          6m20s
csi-azuredisk-controller-849d854b96-lkplx   5/5     Running   0          6m20s
csi-azuredisk-node-7qwlj                    3/3     Running   6          6m20s
csi-azuredisk-node-8xm4c                    3/3     Running   6          6m20s
csi-azuredisk-node-snsdb                    3/3     Running   6          6m20s
```

**get storage class**
```
% kubectl get sc        
NAME                             PROVISIONER          RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
sat-azure-block-bronze           disk.csi.azure.com   Delete          Immediate              true                   164m
sat-azure-block-bronze-metro     disk.csi.azure.com   Delete          WaitForFirstConsumer   true                   164m
sat-azure-block-gold             disk.csi.azure.com   Delete          Immediate              true                   164m
sat-azure-block-gold-metro       disk.csi.azure.com   Delete          WaitForFirstConsumer   true                   164m
sat-azure-block-platinum         disk.csi.azure.com   Delete          Immediate              true                   164m
sat-azure-block-platinum-metro   disk.csi.azure.com   Delete          WaitForFirstConsumer   true                   164m
sat-azure-block-silver           disk.csi.azure.com   Delete          Immediate              true                   164m
sat-azure-block-silver-metro     disk.csi.azure.com   Delete          WaitForFirstConsumer   true                   164m
```

**get secret**
```
% kubectl get secret azure-cloud-provider -n kube-system
NAME                   TYPE     DATA   AGE
azure-cloud-provider   Opaque   1      166m
```

**get PVC**
```
% kubectl get pvc                                    
NAME            STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS                 AGE
pvc-azuredisk   Bound    pvc-eb037445-b832-4cd1-8c34-fb6125c00832   10Gi       RWO            sat-azure-block-gold-metro   17s
```

**get pod which attaches the above PVC**
```
% kubectl get pods               
NAME              READY   STATUS    RESTARTS   AGE
nginx-azuredisk   1/1     Running   0          27s
```

**Verify if the data is being written**
```
% kubectl exec -it nginx-azuredisk -- tail -f /mnt/azuredisk/outfile                   
Tue Jun 22 06:16:22 UTC 2021
Tue Jun 22 06:16:23 UTC 2021
Tue Jun 22 06:16:24 UTC 2021
Tue Jun 22 06:16:25 UTC 2021
Tue Jun 22 06:16:26 UTC 2021
Tue Jun 22 06:16:27 UTC 2021
Tue Jun 22 06:16:28 UTC 2021
Tue Jun 22 06:16:29 UTC 2021
Tue Jun 22 06:16:30 UTC 2021
Tue Jun 22 06:16:31 UTC 2021
Tue Jun 22 06:16:32 UTC 2021
Tue Jun 22 06:16:33 UTC 2021
```

### CODE REVIEW

File: config-templates/azure/azuredisk-csi-driver/1.4.0/README.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+
+
+## Creating the AZURE DISK CSI Driver storage configuration
+
+**Example `sat storage config create` command**
+
+```sh
+ibmcloud sat storage config create --name <config-name> --template-name azuredisk-csi-driver --template-version 1.4.0 -p "cloud-config=<erialized value created from azure.json file>"
+```
```
Comment (nkkashyap) [inline config-templates/azure/azuredisk-csi-driver/1.4.0/README.md:68]: pls add `--location <location>` option

**Review by nkkashyap — APPROVED** [review]
LGTM
A minor comment in readme

File: config-templates/azure/azuredisk-csi-driver/1.4.0/README.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
+# AZURE DISK CSI Driver[BETA]
+
+AZURE DISK CSI driver implements the CSI specification for container orchestrators to manage the lifecycle of Azure DISK volumes.
+
+**Features supported:**
+- Topology(Availability Zone)
+    - ZRS disk support(Preview)
```
Comment (derekpoindexter) [inline config-templates/azure/azuredisk-csi-driver/1.4.0/README.md:7]:
```suggestion
- ZRS disk support(Preview)
```

File: config-templates/azure/azuredisk-csi-driver/1.4.0/custom-parameters.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+[
+    {
+        "description": "cloud-config : This is a base64 encoded string generated from azure.json",
```
Comment (derekpoindexter) [inline config-templates/azure/azuredisk-csi-driver/1.4.0/custom-parameters.json:3]:
```suggestion
        "description": "cloud-config : The base64 encoded string generated from the azure.json config file.",
```

File: config-templates/azure/azuredisk-csi-driver/1.4.0/metadata.json [hunk 4]
```diff
@@ -0,0 +1,63 @@
+{
+    "name": "1.4.0",
+    "enabled": "true",
+    "description": "AZURE DISK CSI Driver version 1.4.0[BETA]",
```
Comment (derekpoindexter) [inline config-templates/azure/azuredisk-csi-driver/1.4.0/metadata.json:4]:
```suggestion
    "description": "[Beta] - Azure Disk CSI Driver",
```

File: config-templates/template_list.json [hunk 5]
```diff
@@ -15,6 +15,14 @@
     "enabled": "true",
     "provider": "aws"
   },
+  {
+    "name": "azuredisk-csi-driver",
+    "displayname": "AZUREDISK CSI Driver",
```
Comment (derekpoindexter) [inline config-templates/template_list.json:20]:
```suggestion
    "displayname": "Azure Disk CSI Driver",
```

File: config-templates/template_list.json [hunk 6]
```diff
@@ -15,6 +15,14 @@
...
+    "Description": "AZUREDISK CSI Driver[BETA].",
```
Comment (derekpoindexter) [inline config-templates/template_list.json:21]:
```suggestion
    "Description": "Azure Disk CSI Driver",
```

**Review by derekpoindexter — CHANGES_REQUESTED** [review]

File: config-templates/azure/azuredisk-csi-driver/1.4.0/metadata.json [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+            "name": "support-msg",
+            "value": "For support contact Microsoft Azure"
+        },
+        {
+            "name": "provider-logo",
+            "value": "logo/azuredisk-csi-driver.png"
+        },
+        {
+            "name": "supported-envs",
+            "value": "AZURE"
```
Comment (derekpoindexter) [inline config-templates/azure/azuredisk-csi-driver/1.4.0/metadata.json:28]:
```suggestion
            "value": "Azure"
```

**Review by derekpoindexter — APPROVED** [review]

**Pull request merged**